### ISSUE TITLE
Progressive mode doesn't report proper exit code in ansible-lint 6.8.2

### ISSUE THREAD

**Issue by xfix** [ansible/ansible-lint#2613]
##### Summary

Progressive mode is broken when there is at least one error. This is a regression introduced in ansible-lint 6.8.2. I think it was introduced by https://github.com/ansible/ansible-lint/commit/388d015621527b407fb1d08f67c6f7b60f00498e.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible [core 2.13.5]
  config file = None
  configured module search path = ['/home/xfix/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-M8rP2D-G-py3.10/lib/python3.10/site-packages/ansible
  ansible collection location = /home/xfix/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/xfix/.cache/pypoetry/virtualenvs/ansible-lint-bug-report-M8rP2D-G-py3.10/bin/ansible
  python version = 3.10.7 (main, Sep  5 2022, 13:12:31) [GCC 11.3.0]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version

ansible-lint 6.8.2 using ansible 2.13.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

NixOS unstable, Python 3.10, installed using Poetry

```
[tool.poetry.dependencies]
python = "^3.9"
ansible-lint = "^6.8.2"
ansible = "^6.5.0"
```

##### STEPS TO REPRODUCE

Run `ansible-lint --progressive` in repository that has at least one problem that wasn't introduced in the most recent commit. For an example of such repository, clone `https://gist.github.com/ce4dec8ad7192ba9b08502d64e29a0f8.git`.

##### Desired Behavior

`--progressive` should return `0` status code if no new issues were introduced.

##### Actual Behavior

I get status code `2` after running `ansible-lint --progressive`.

```
$ poetry run ansible-lint --progressive
WARNING  Total violations not increased since previous commit, will mark result as success. (1 -> 0)
WARNING  Marked 1 previously known violation(s) as ignored due to progressive mode.
WARNING  Listing 1 violation(s) marked as ignored, likely already known
yaml[colons]: Too many spaces after colon
blah.yml:1


             Rule Violation Summary              
 count tag          profile rule associated tags 
     1 yaml[colons] basic   formatting, yaml     

Failed after min profile: 1 failure(s), 0 warning(s) on 4 files.
$ echo $?
2
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
